packages/xo-server/src/xo-mixins/plugins.js

```diff
@@ -154,12 +154,14 @@ export default class {
 
     // Sets the plugin configuration.
     await plugin.instance.configure(
-      {
-        // Shallow copy of the configuration object to avoid most of the
-        // errors when the plugin is altering the configuration object
-        // which is handed over to it.
-        ...configuration,
-      },
+      // Shallow copy of the configuration object to avoid most of the
+      // errors when the plugin is altering the configuration object
+      // which is handed over to it.
+      Array.isArray(configuration)
+        ? [...configuration]
+        : {
+            ...configuration,
+          },
```

Comment:
1. Why this change?
2. Is it really a good idea to allow array configurations?
3. If yes, wouldn't it be better to simply freeze `configuration`?

Reply:
1. Because the web-hooks plugin has an array configuration
2. Why wouldn't it be?
3. Why not but why would it be better?

Reply:
2. I think it's a bad idea, the array should be in a top level object in case you want to add a global config in the future
3. I was thinking about a deep freeze, sorry for being unclear, because it would properly ensure that the whole object is not mutated no matter its type.

Reply:
2. Right, I agree with that actually, I'll change it in the plugin.